src/edu/csus/ecs/pc2/shadow/RemoteRunSubmitter.java

```diff
@@ -162,7 +162,8 @@ private Account getSubmitterId(IInternalContest internalContest, String clientId
         // search for this site's logins first
         for (Account account : accounts) {
             if (account.getSiteNumber() == thisSiteNumber) {
-                if (account.getClientId().getName().equals(clientIdString)) {
+                if (account.getClientId().getName().equals(clientIdString) ||
+                    account.getAliasName().equals(clientIdString)) {
```

Comment:
Hmmm.... I'm suspicious of the new call to `getAliasName()`.    Indeed, PC2 provides for an alternative account name, called an "alias"; the "alias" is intended to be used to support the ability to display a meaningful "name" to a judge without allowing the judge to actually know what (real) team they are working with (thus protecting the judge from allegations of "favoring" a particular team because the judge knew which team he/she was working on).  

For example, "UC Berkeley" can be "aliased" to some non-identifying name such as "TeamWithPythonIssue"  or something more generic like "Team B";  thereafter, any time the (human) judge sees a team name on a submission it will appear as "TeamWithPythonIssue" (or "Team B") instead of "UC Berkeley".   

Displaying (or not) of "aliases" is controlled in the Admin>ConfigureContest>Settings tab, under "Judge Settings>Team Information Displayed to Judges"  (by selecting the "Show Alias" radio button).

Note that a team "alias" is NOT the same thing as a team "Display Name".  If you select "Accounts>Edit Account" on the Admin, you'll see that a team has both an "Account"  (e.g. "team500"), a "Display Name" (which _defaults_ to the same thing as the "account" _but can be changed to any value_.)  It is the "Display Name" which appear on the scoreboard.  

But accounts also have a separate "Alias" field, which can also be set to any value.  The "alias" value is only used (as far as I can recall) in the above-described "Judge Settings".

Sooo... I have a strong suspicion that the above addition referencing `account.getAliasName()` is actually intended to call `account.getDisplayName()`.  Please either confirm this and make the change, or explain why `getAliasName()` is being called.

Reply:
The call to `getAliasName()` is what was intended.  After a discussion with @clevengr we are on the same page as to how this change works.  The idea is to allow an arbitrary **team_id** (such as '**domjudge**') to be matched.  Normally this is not possible since all team accounts are always **team###**.  By using the _alias_ mechanism, we can set the _alias_ to an arbitrary string and the remote submitter can then lookup either an expected **team_id** (a number such as 521) or an arbitrary string (using the _alias_).  There was some discussion about using the _display name_ instead of _alias_, but the issue there is we wanted to keep the _display name_ as a meaningful string that conveyed what the account was for, eg. "_DOMJudge Jury Test Team_".  Note that all normal team accounts (**team###**) will have an empty _alias_ field, and this change will not affect the lookup of those team accounts.  Only the special test accounts would have an _alias_ manually defined by the administrator based on what the remote CCS uses for team test account(s).

Reply:
> Hmmm.... I'm suspicious of the new call to getAliasName(). Indeed, PC2 provides for an alternative account name, called an "alias"; the "alias" is intended to be used to support the ability to display a meaningful "name" to a judge without allowing the judge to actually know what (real) team they are working with (thus protecting the judge from allegations of "favoring" a particular team because the judge knew which team he/she was working on).

The original intention for alias name was as @johnbrvc and I discussed, it was to
allow a user to login as judge1 or as jo.   Later, it was then used/leveraged as an option
in the Team Information Displayed to Judges.

In both cases the alias is used as an alias for the team.   Change approved.

Reply:
I still think this is somewhat problematic -- now we have _triply_ overloaded the meaning and use of "alias".  I see the possibility that this use of the same field for three different purposes might come back to bite us later.

However, having said that, after conversations with @johnbrvc I agree that the change will work here as intended, and given the _extremely_ small number of times that `alias` gets used for anything (at least, as far as we're aware), it's probably not that big an issue and certainly can be dealt with later.  In any case, it's certainly not worth holding up the PR (which is why I "approved" the PR earlier, even with this change present).  I'll mark the comment "resolved".